arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -823,7 +823,9 @@ Result RocksDBCollection::insert(arangodb::transaction::Methods* trx,
     }
   }
 
-  LocalDocumentId const documentId = LocalDocumentId::create();
+  bool useRev = _logicalCollection.version() >= LogicalCollection::Version::v36;
+  LocalDocumentId const documentId =
+      useRev ? LocalDocumentId::create(revisionId) : LocalDocumentId::create();
```

Comment:
I think it is effectively not an issue to unconditionally use the revision (except in the smart edge case)

LocalDocumentId and the revisions are generated from the same server-global HLC counter

Reply:
i.e. TRI_HybridLogicalClock

Reply:
I would agree, except that it seems that there is no good way to reliably detect if we are in a smart edge child collection prior to 3.6 (and no way to do so on upgrade either). Hence, limiting to 3.6 or higher.

Reply:
but the revision in the smart edge child will only use the cluster wide unique revisions if the collection was created with 3.6 no ?

Reply:
another question: can we determine if all the shards part of a smart edge collection were created with 3.6 ? 
Because I fear that we are not able to effectively differentiate whether all the shards are using the unique ID

Reply:
@graetzer Yes, it will only use the cluster wide unique revisions if created with 3.6. However, if it doesn't use this value, then it can't use the revisionId from the document as the LocalDocumentId. It should instead use an actual local ID.

Hm. This question about matching all shard versions could be more troublesome. Let's say we have an old-style collection created in 3.4 or whatever. Then we upgrade to 3.6. How do we create a new shard, I guess by increasing replication factor or something? I think that in this case, it should still get the collection version from the creation slice, which should be an exact clone of the old leader shard, right? So it would still be tagged 3.4?

Reply:
the problem is the agency does not contain the collection version

Reply:
@graetzer Wait, it doesn't? So whenever we create a new follower shard, we populate the info anew on the DBServer or something? In that case, this entire plan is pretty screwed.

Reply:
For the community case this is entirely sufficient no ? 
For the enterprise smart edge case this is an issue, because we need to be consistent across all three shards (i.e. _from, _to and _local)